### ISSUE TITLE
Cannot update these three packages

### ISSUE THREAD

**Issue by DarthSonic** [microsoft/winget-cli#3038]
### Brief description of your issue

![image](https://user-images.githubusercontent.com/3968605/222688109-0f652e30-1197-4883-8f3c-845fd1b2593d.png)

- Azure CLI is already 2.45 on my system.
- JetBrains ETW Host Service (x64) is 16.34.0 (whole different syntax of version) but ReSharper is 2022.3.2 (which would be higher anyway).
- Windows SDK is installed 10.1.20348.1 and 10.1.22621.755, both higher than what winget seems to find.

All installations will run but do not update anything.

### Steps to reproduce

winget upgrade

### Expected behavior

do not show packages as outdated

### Actual behavior

shows packages as outdated

### Environment

```shell
Windows Package Manager v1.3.2691
Windows 11 Pro 22H2 22623.1325
```

**Comment by stephengillie** [comment]
Hi @DarthSonic,

All of these seem to be showing an older version than what you have installed. Is that accurate? Could these be from old registry entries that may have been left behind from previous installations?

**Comment by DarthSonic** [comment]
> old

Yes. All show lower versions than installed. All are currently installed. I can't say if there are registry keys left over. The tools were updated by their own updater or by downloading new version.
